src/store/fileUploadStore.js

```diff
@@ -108,44 +112,36 @@ const mutations = {
 }
 
 const actions = {
-
-	/**
-	 *
-	 * @param {object} context The context object
-	 * @param {string} token The conversation's token
-	 * @param {array} files The files to be processed and added to the store
-	 */
-	addFilesToBeUploaded(context, { token, files }) {
-		files.forEach(file => {
-			context.commit('addFileToBeUploaded', { token, file })
-		})
-	},
-
 	/**
 	 * Uploads the files to the root directory of the user
 	 * @param {object} param0 Commit, state and getters
-	 * @param {*} token The conversation's token
+	 * @param {object} param1 The unique uploadId, the conversation token and the
+	 * files array
 	 */
-	async uploadFiles({ commit, state, getters }, token) {
+	async uploadFiles({ commit, state, getters }, { uploadId, token, files }) {
+		files.forEach(file => {
+			commit('addFileToBeUploaded', { uploadId, token, file })
+		})
 		// Iterate through the previously indexed files for a given conversation (token)
-		for (const index in state.files[token]) {
-			if (state.files[token][index].status !== 'toBeUploaded') {
-				continue
-			}
+		for (const index in state.uploads[uploadId].files) {
 			// Mark file as uploading to prevent a second function call to start a
 			// second upload for the same file
-			commit('markFileAsUploading', { token, index })
+			commit('markFileAsUploading', { uploadId, index })
 			// Get the current user id
 			const userId = getters.getUserId()
 			// currentFile to be uploaded
-			const currentFile = state.files[token][index].file
+			const currentFile = state.uploads[uploadId].files[index].file
 			// Destination path on the server
 			const path = '/files/' + userId + getters.getAttachmentFolder() + '/' + currentFile.name
+			const uniquePath = await findUniquePath(client, path)
 			try {
 				// Upload the file
-				await client.putFileContents(path, currentFile)
+				await client.putFileContents(uniquePath, currentFile)
+				// Compute the sharePath by removing the first part of the upload path
+				const partOfPathToRemove = '/files/' + userId
+				const sharePath = uniquePath.substring(partOfPathToRemove.length, uniquePath.length)
```

Comment:
make your life easier by not passing `'/files/' + userId` to `findUniquePath` in first place?

Reply:
This has to be done, whether it's in findUniquePath or outside of it. So no easier life I think :)

Reply:
I ment something like https://gist.github.com/nickvergessen/e88c72d68fbc5d457a2c145c9e880954